components/suggest/src/db.rs

```diff
@@ -1192,6 +1196,22 @@ impl<'a> SuggestDao<'a> {
         Ok(())
     }
 
+    pub fn insert_dismissal(&self, url: &str) -> Result<()> {
+        self.conn.execute(
+            "INSERT OR IGNORE INTO dismissed_suggestions(url_hash)
+             VALUES(hash(:url))",
+            named_params! {
+                ":url": url,
+            },
+        )?;
+        Ok(())
+    }
+
+    pub fn clear_dismissals(&self) -> Result<()> {
```

Comment:
```suggestion
    pub fn clear_dismissals(&mut self) -> Result<()> {
```

(Just a loose convention that we've been using for the other DAO methods!)

Reply:
Good catch, I like those conventions.